sympy/functions/special/delta_functions.py

```diff
@@ -159,6 +159,26 @@ def is_simple(self, x):
             return p.degree() == 1
         return False
 
+    def _eval_rewrite_as_Piecewise(self, arg):
+        """Represents DiracDelta in a Piecewise form
+
+           Examples
+           ========
+
+           >>> from sympy import DiracDelta, Piecewise, Symbol
+           >>> from sympy.abc import x, y
+           >>> x = Symbol('x', real=True)
+
+           >>> DiracDelta(x).rewrite(Piecewise)
+           Piecewise((undefined, Eq(x, 0)), (0, True))
+
+           >>> DiracDelta(x - 5).rewrite(Piecewise)
+           Piecewise((undefined, Eq(x - 5, 0)), (0, True))
+
+        """
+        if arg.is_real:
+            return Piecewise(('undefined', Eq(arg, 0)), (0, True))
```

Comment:
Thanks @asmeurer for the review. I will make it `oo`.
And I will keep in mind about the way to use a string as input.